src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Serialization/FullProjectSnapshotHandleJsonConverter.cs

```diff
@@ -26,30 +26,65 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                 return null;
             }
 
-            var obj = JObject.Load(reader);
+            string serializationFormat = null;
+            string filePath = null;
+            RazorConfiguration configuration = null;
+            string rootNamespace = null;
+            ProjectWorkspaceState projectWorkspaceState = null;
+            DocumentSnapshotHandle[] documents = null;
+
+            reader.ReadProperties(propertyName =>
+            {
+                switch (propertyName)
+                {
+                    case SerializationFormatPropertyName:
```

Comment:
What behavior makes sense for encountering unexpected properties? Should have a default case for that.

Reply:
IMO in the interest of extensibility (and future backwards compatibility) we should just ignore unexpected properties. We're being pretty careful of what properties are written in `WriteJson` so this hopefully doesn't pose an issue.